src/plugins/visualizations/server/saved_objects/visualization_migrations.ts

```diff
@@ -981,6 +982,96 @@ const removeDefaultIndexPatternAndTimeFieldFromTSVBModel: SavedObjectMigrationFn
   return doc;
 };
 
+const addEmptyValueRuleForSavedObjectsWithLessAndGreaterThenZeroRules: SavedObjectMigrationFn<
+  any,
+  any
+> = (doc) => {
+  const visStateJSON = get(doc, 'attributes.visState');
+  let visState;
+
+  if (visStateJSON) {
+    try {
+      visState = JSON.parse(visStateJSON);
+    } catch (e) {
+      // Let it go, the data is invalid and we'll leave it as is
+    }
+
+    if (visState && visState.type === 'metrics') {
+      const params: any = get(visState, 'params') || {};
+      const {
+        bar_color_rules: barColorRules = [],
+        background_color_rules: backgroundColorRules = [],
+        gauge_color_rules: gaugeColorRules = [],
+      } = params;
+
+      const compareWithEqualMethods = ['lte', 'gte'];
+      const emptyRule = {
+        operator: 'empty',
+        value: null,
+      };
+
+      const getRulesWithComparingToZero = (rules: any[] = []) =>
+        last(
+          rules.filter(
+            (rule) => compareWithEqualMethods.includes(rule.operator) && rule.value === 0
+          )
+        );
+
+      const convertRuleToEmpty = ({ id, ...rest }: any = {}) => ({
+        ...rest,
+        ...emptyRule,
+        id: uuid.v4(),
+      });
+
+      const lastBarRule = getRulesWithComparingToZero(barColorRules) ?? null;
+      const lastBackgroundRule = getRulesWithComparingToZero(backgroundColorRules) ?? null;
+      const lastGaugeRule = getRulesWithComparingToZero(gaugeColorRules) ?? null;
+
+      const colorRules = {
+        bar_color_rules: barColorRules,
+        background_color_rules: backgroundColorRules,
+        gauge_color_rules: gaugeColorRules,
+      };
+
+      if (lastBarRule) {
```

Comment:
nit: maybe to avoid this type of duplication we can create a separate method for that?  

```
const addEmptyRule = (rules, key) => {
  const rulesWithComparingToZero = getRulesWithComparingToZero(barColorRules);

  if (rulesWithComparingToZero) {
    colorRules[key] = [...colorRules[key], convertRuleToEmpty(rulesWithComparingToZero)];
  }
};
```

Reply:
We are modifying some external array in this case... As for me, such mutations are very dangerous, because it is difficult to recognize, what is going wrong with the external data structure. Exactly that's why I was modifying the array last time on every step.

Reply:
I've taken your advice and it looks much better right now. Thanks)